### PR TITLE
Refactors decorators to live in a separate subpackage

### PR DESCRIPTION
**Pull request by elijahbenizzy**
This makes it backwards compatible -- we previously had two files:

1. function_modifiers.py
2. function_modfiers_base.py

We now create a module `function_modifiers.py` which enables us to separate out decorators, as the file is getting unwieldy and ugly.

That said, we need to maintain backwards compatibility on import. To do this, we:

1. Create a function_modifiers subpackage
2. Put everything that was in function_modifiers.py in __init__.py in that package (so imports work)
3. Move function_modifiers_base.py over there
4. Create a "dummy" function_modifiers_base.py that just imports everything from the actual one.

Will need to test this out on sample workflows prior to publishing.

[Short description explaining the high-level reason for the pull request]

## Changes

-

## Testing

1.

## Checklist

- [ ] PR has an informative and human-readable title (this will be pulled into the release notes)
- [ ] Changes are limited to a single goal (no scope creep)
- [ ] Code can be automatically merged (no conflicts)
- [ ] Code passed the pre-commit check & code is left cleaner/nicer than when first encountered.
- [ ] Passes all existing automated tests
- [ ] Any _change_ in functionality is tested
- [ ] New functions are documented (with a description, list of inputs, and expected output)
- [ ] Placeholder code is flagged / future TODOs are captured in comments
- [ ] Project documentation has been updated if adding/changing functionality.
- [ ] Reviewers requested with the [Reviewers tool](https://help.github.com/articles/requesting-a-pull-request-review/) :arrow_right:

### Python - local testing

- [ ] python 3.6
- [ ] python 3.7

### CODE REVIEW

**Comment by elijahbenizzy** [comment]
Comments from @skrawcz:

(originally here): https://github.com/stitchfix/hamilton/pull/199#pullrequestreview-1112603506

I'm concerned about:

1. commit history preservation - wouldn't want to lose some of the "why" for some of the decorators.
2. Any reason function_modifiers_base needs to exist? Where is it being used? Can we add a deprecation warning if we can't migrate things ourselves?
3. Shouldn't there be a base.py in the function_modifiers module now?

+ 2 code comments:
- https://github.com/stitchfix/hamilton/pull/199#discussion_r974491540
- https://github.com/stitchfix/hamilton/pull/199#discussion_r974494130

File: tests/function_modifiers/test_configuration.py [hunk 1]
```diff
@@ -0,0 +1,84 @@
+import pytest
+
+import hamilton.function_modifiers
+from hamilton import function_modifiers, function_modifiers_base
```
Comment (skrawcz) [inline tests/function_modifiers/test_configuration.py:4]: can remove `function_modifiers_base`?

File: tests/function_modifiers/test_expanders.py [hunk 2]
```diff
@@ -0,0 +1,491 @@
+from typing import Any, Dict
+
+import numpy as np
+import pandas as pd
+import pytest
+
+import hamilton.function_modifiers
+from hamilton import function_modifiers, node
+from hamilton.function_modifiers import source, value
```
Comment (skrawcz) [inline tests/function_modifiers/test_expanders.py:9]: feels like this can be tidied up

File: tests/function_modifiers/test_macros.py [hunk 3]
```diff
@@ -0,0 +1,227 @@
+import inspect
+from typing import List, Set
+
+import pandas as pd
+import pytest
+
+import hamilton.function_modifiers
+from hamilton import function_modifiers, models
+from hamilton.function_modifiers import does
+from hamilton.function_modifiers.macros import ensure_function_empty
```
Comment (skrawcz) [inline tests/function_modifiers/test_macros.py:10]: ditto

File: hamilton/function_modifiers/__init__.py [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+import logging
+from typing import Any, Callable, Collection, Dict, List, Type
+
+from hamilton import node
+
+from . import (
+    configuration,
+    dependencies,
+    expanders,
+    function_modifiers_base,
```
Comment (skrawcz) [inline hamilton/function_modifiers/__init__.py:11]: can rename this to `base`?

**Pull request closed without merging**